src/providers/realtime-provider.js

```diff
@@ -113,15 +167,29 @@ export default class RealtimeEventsProvider {
         this.socket.onmessage = (event) => {
             let data = JSON.parse(event.data);
 
-            if (data.type === MESSAGES.DATA_TYPE_REPLY) {
+            if (data.type === DATA_TYPES.DATA_TYPE_REPLY) {
                 let replyToId = data.data.replyTo;
                 let subscriptionDetails = this.getSubscriptionDetailsById(replyToId);
                 subscriptionDetails.call = data.call;
-            } else if (data.type === MESSAGES.DATA_TYPE_EVENTS) {
+            } else if (this.isSupportedDataType(data.type)) {
                 let call = data.call;
                 let subscriptionDetails = this.getSubscriptionDetailsByCall(call);
 
-                subscriptionDetails.callback(data.data);
+                if (data.type === DATA_TYPES.DATA_TYPE_EVENTS) {
+                    subscriptionDetails.callback(data.data);
+                } else if (subscriptionDetails && data.data.values) {
+                    for (let i = 0; i < data.data.values.length; i++) {
```

Comment:
`data.data.values.forEach(() => {....});`

Reply:
I almost did that, but this is getting called... up to MANY times a second and I figured a straight for loop may be slightly faster rather than invoking a callback. Thoughts?

Reply:
In fact, I may want to cache the length...

Reply:
Smort. I'd be curious to see a benchmark of `for` vs. `.forEach` but I agree with you, generally speaking invoking functions is more expensive than not.

Reply:
In fact I am setting paramater as well, so I'll just do as you suggested :)